examples/sssp/sssp.py

```diff
@@ -0,0 +1,130 @@
+        for i in range(len(edge_list)):
+            self.local_graph[edge_list[i][0] - self.start_vertex][1].append((edge_list[i][1], edge_list[i][2]))
+        self.reduce(callback, None, Reducer.nop)
+    
+    def calculate_destination(self, vertex_index):
+        for i in range(len(self.partition_indices)-1):
+            if vertex_index >= self.partition_indices[i] and vertex_index < self.partition_indices[i+1]:
+                return i
+        return len(self.partition_indices)-1
+    
+    def update_distance(self, update):
+        local_index = update[0]-self.start_vertex
+        if update[1] < self.local_graph[local_index][2]:
+            self.local_graph[update[0]-self.start_vertex][2] = update[1]
+            for i in range(len(self.local_graph[local_index][1])):
+                dest_vertex = self.local_graph[local_index][1][i][0]
+                dest_partition = self.calculate_destination(dest_vertex)
+                cost = self.local_graph[local_index][2] + self.local_graph[local_index][1][i][1]
+                new_update = (dest_vertex, cost)
+                self.thisProxy[dest_partition].update_distance(new_update)
+    
+    def print_results(self, callback):
+        max_local_cost = 0.0
+        for i in range(len(self.local_graph)):
+            #print("Final cost of vertex", self.local_graph[i][0], ":", self.local_graph[i][2])
+            if self.local_graph[i][2] > max_local_cost:
+                max_local_cost = self.local_graph[i][2]
+        self.reduce(callback, max_local_cost, Reducer.max)
+
+
+class Main(Chare):
+
+    def __init__(self, args):
+        if len(args) != 5:
+            print("Wrong number of arguments. Usage: sssp.py <num_vertices> <num_edges> <random_seed> <source_vertex>")
+            exit()
+        #define parameters
+        self.num_vertices = int(args[1])
+        self.num_edges = int(args[2])
+        self.random_seed = int(args[3])
+        self.source_vertex = int(args[4])
+        if self.source_vertex < 0 or self.source_vertex > self.num_vertices-1:
+            print("Source vertex out of range")
+            exit()
+        #generate edges randomly and sort them by edge source
+        begin_generation = time.time()
+        random.seed(self.random_seed)
+        self.edge_list = []
+        for i in range(self.num_edges):
+            edge_source = random.randint(0, self.num_vertices-1)
+            edge_dest = random.randint(0, self.num_vertices-1)
+            while edge_source==edge_dest:
+                edge_dest = random.randint(0, self.num_vertices-1)
+            edge_weight = random.random()
+            self.edge_list.append((edge_source, edge_dest, edge_weight))
+        self.edge_list.sort(key=lambda a: a[0])
+        #initiate worker array
+        num_partitions = charm.numPes()
+        self.workers = Array(SsspChares, num_partitions)
+        charm.awaitCreation(self.workers)
```

Comment:
Why is this call necessary?

Reply:
From what I saw in other examples, I thought this was necessary to start calling entry methods on the chare array. Some testing now shows that it's not needed, so I'll delete it.